File: python_cpp/uPIMulator_frontend/src/converter/endian_converter.py
from abi.isa.instruction.endian import Endian


class EndianConverter:
    def __init__(self):
        pass

    @staticmethod
    def convert_to_endian(endian: str) -> Endian:
        if endian == "!little":
            return Endian.LITTLE
        elif endian == "!big":
            return Endian.BIG
        else:
            raise ValueError

    @staticmethod
    def convert_to_string(endian: Endian) -> str:
        if endian == Endian.LITTLE:
            return "!little"
        elif endian == Endian.BIG:
            return "!big"
        else:
            raise ValueError

File: python_cpp/uPIMulator_backend/src/converter/endian_converter.cc
#include "converter/endian_converter.h"

#include <stdexcept>

namespace upmem_sim::converter {

std::string EndianConverter::to_string(abi::isa::Endian endian) {
  if (endian == abi::isa::LITTLE) {
    return "!little";
  } else if (endian == abi::isa::BIG) {
    return "!big";
  } else {
    throw std::invalid_argument("");
  }
}

}  // namespace upmem_sim::converter